test/results/utils/test_protein_shape_file_gen.py

```diff
@@ -0,0 +1,146 @@
+            "Only_Main_Chain",
+        ),
+        # Second
+        (
+            [1, 0, 3, 2],
+            [None, None, 3, 3, None],
+            "APRLR",
+            ["", "", "F", "Y", ""],
+            [
+                None,
+                None,
+                np.array([0.57735027, 0.57735027, -1.73205081]),
+                np.array([2.30940108, -1.15470054, 0.0]),
+                None,
+            ],
+            np.array(
+                [
+                    [0.0, 0.0, 0.0],
+                    [0.57735027, 0.57735027, -0.57735027],
+                    [1.15470054, 0.0, -1.15470054],
+                    [1.73205081, -0.57735027, -0.57735027],
+                    [2.30940108, 0.0, 0.0],
+                ]
+            ),
+            np.array(
+                [
+                    ["A", "0.0", "0.0", "0.0"],
+                    ["P", "0.5773502691896258", "0.5773502691896258", "-0.5773502691896258"],
+                    ["R", "1.1547005383792517", "0.0", "-1.1547005383792517"],
+                    ["L", "1.7320508075688776", "-0.5773502691896258", "-0.5773502691896258"],
+                    ["R", "2.3094010767585034", "0.0", "0.0"],
+                    ["F", "0.5773502691896258", "0.5773502691896258", "-1.7320508075688776"],
+                    ["Y", "2.3094010767585034", "-1.1547005383792517", "0.0"],
+                ],
+                dtype="<U32",
+            ),
+            "Also_Side_Chains",
+        ),
+    )
+    def test_shape(
+        self,
+        main_chain_turns,
+        side_chain_turns,
+        main_chain_residue_sequence,
+        side_chain_residue_sequences,
+        side_positions,
+        main_positions,
+        xyz,
+        name_file,
+    ):
+        """Tests if ProteinShapeFileGen is properly initialized and its attributes are properly set."""
+        peptide = Peptide(
+            main_chain_residue_sequence=main_chain_residue_sequence,
+            side_chain_residue_sequences=side_chain_residue_sequences,
+        )
+        filegen = ProteinShapeFileGen(
+            main_chain_turns=main_chain_turns,
+            side_chain_turns=side_chain_turns,
+            peptide=peptide,
+        )
+        with self.subTest("Side Positions"):
+            for result, expected in zip(filegen.side_positions, side_positions):
+                if expected is None:
+                    self.assertIsNone(result)
+                else:
+                    np.testing.assert_almost_equal(result, expected, decimal=6)
+
+        with self.subTest("Main Positions"):
+            np.testing.assert_almost_equal(
+                filegen.main_positions,
+                main_positions,
+                decimal=6,
+            )
+        with self.subTest("XYZ file data"):
+            np.testing.assert_equal(filegen.get_xyz_data(), xyz)
+
+        with self.subTest("Write file"):
+            current_dir = os.path.dirname(__file__)
+            test_path = os.path.join(current_dir, "resources")
```

Comment:
I agree with Steve's previous comment, that you should be using `tempfile` to determine the location where this file goes.
This will also prevent leaving around left-over files, in case the test below fails (in which case the `os.remove` call would never happen).

See here for an existing example: https://github.com/Qiskit/qiskit-nature/blob/cf61e81473c97b5f7b290e24a02ade2c1b08a9ac/test/drivers/second_quantization/fcidumpd/test_driver_fcidump_dumper.py#L132